tlatools/org.lamport.tlatools/test/util/ParserAPI.java

```diff
@@ -130,7 +129,6 @@ public static ModuleNode processSemantics(SyntaxTreeNode parseTree, Errors log)
    * @return Whether levels in parse tree are correct.
    */
   public static boolean checkLevel(ModuleNode semanticRoot, Errors log) {
```

Comment:
Is it now possible to refactor this into a method of ModuleNode?

Reply:
By the way, (in a separate PR) the function `processSemantics` could be moved into `ModuleNode` as e.g. `ModuleNode#create`, `processSyntax` could be moved into `SyntaxTreeNode`, and `parse` could be moved into `SANY`.

Reply:
It is a trivial enough wrapper here that `ParserAPI.checkLevel` could be done away with, but if we are designing the `ParserAPI` class to eventually be the primary API through which developers use SANY, it makes sense to expose `checkLevel` as one of a small number of functions in this class for discoverability purposes - otherwise, consumers would have to know about the `ModuleNode.checkLevel` method. It's just a discoverability issue for such an important function.

Reply:
Personally, I don’t find a utility class discoverable, but it’s pointless to discuss discoverability since we lack an objective measure.

Reply:
I'm envisioning this class would eventually graduate to living in `src/tla2sany/API.java` or similar, which might help.